File: src/gestorAplicacion/Medicamentos.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package gestorAplicacion;
import gestorAplicacion.personas.Paciente;
import java.util.ArrayList;
/**
 *
 * @author Roger Vera
 */
public class Medicamentos {
    private tipoMedicamento tipoMed;
    private Paciente paciente;
    private int dosis;//cantidad a consumir
    private int modoUso;//consumir cada x horas
    private ArrayList<Medicamentos> medicamiento = new ArrayList<>();
    
    public static int cantAcetaminofen=100;
    public static int cantAmoxicilina=100;
    public static int cantIbuprofeno=100;
    public static int cantLagrimasArtificiales=20;
    
    //constructor
    public Medicamentos(tipoMedicamento tipoMed, Paciente paciente, int dosis, int modoUso) {
        this.tipoMed = tipoMed;
        this.paciente = paciente;
        this.dosis = dosis;
        this.modoUso = modoUso;
        medicamiento.add(this);
    }
    
        
    
    //getter & setter
    public tipoMedicamento getTipoMed() {
        return tipoMed;
    }
    public void setTipoMed(tipoMedicamento tipoMed) {
        this.tipoMed = tipoMed;
    }
    
    public Paciente getPaciente() {
        return paciente;
    }
    public void setPaciente(Paciente paciente) {
        this.paciente = paciente;
    }

    public int getDosis() {
        return dosis;
    }
    public void setDosis(int dosis) {
        this.dosis = dosis;
    }

    public int getModoUso() {
        return modoUso;
    }
    public void setModoUso(int modoUso) {
        this.modoUso = modoUso;
    }

    public ArrayList<Medicamentos> getMedicamiento() {
        return medicamiento;
    }

    public void setMedicamiento(ArrayList<Medicamentos> medicamiento) {
        this.medicamiento = medicamiento;
    }
    
    //metodos
    
    //resta la cantidad de medicamento asignado del inventario de medicamentos
    public static void asignarMed(tipoMedicamento eleccion,int cantidad){
        switch (eleccion) {
            case Acetaminofen:
                cantAcetaminofen=cantAcetaminofen-cantidad;
                break;
            case Amoxicilina:
                cantAmoxicilina=cantAmoxicilina-cantidad;
                break;
            case Ibuprofeno:
                cantIbuprofeno=cantIbuprofeno-cantidad;
                break;
            default:
                cantLagrimasArtificiales=cantLagrimasArtificiales-cantidad;
                break;
        }        
    }

}

File: Python/src/gestorAplicacion/Medicamentos.py
from gestorAplicacion.tipoMedicamento import tipoMedicamento
class Medicamentos:
    cantAcetaminofen =100
    cantAmoxicilina =100
    cantIbuprofeno =100
    cantLagrimasArtificiales =20
    _medicamiento = []

    #constructor
    def __init__(self, tipoMed, paciente, dosis, modoUso):
        self._tipoMed = tipoMed
        self._paciente = paciente
        self._dosis = dosis #cantidad a consumir
        self._modoUso = modoUso #consumir cada x horas
        self._medicamiento.append(self)
    
    #getter & setter
    def getTipoMed(self):
        return self._tipoMed
    def setTipoMed(self, tipoMed):
        self._tipoMed = tipoMed

    def getPaciente(self):
        return self._paciente
    def setPaciente(self, paciente):
        self._paciente = paciente

    def getDosis(self):
        return self._dosis
    def setDosis(self, dosis):
        self._dosis = dosis

    def getModoUso(self):
        return self._modoUso
    def setModoUso(self, modoUso):
        self._modoUso = modoUso

    def getMedicamiento(self):
        return self._medicamiento

    def setMedicamiento(self, medicamiento):
        self._medicamiento = medicamiento

    #metodos
    #resta la cantidad de medicamento asignado del inventario de medicamentos
    @staticmethod
    def asignarMed(eleccion, cantidad):
        if eleccion == tipoMedicamento.ACETAMINOFEN.getTipoMedicamento(): #"ACETAMINOFEN"
            Medicamentos.cantAcetaminofen=Medicamentos.cantAcetaminofen-cantidad
        elif eleccion == tipoMedicamento.AMOXICILINA.getTipoMedicamento():
            Medicamentos.cantAmoxicilina=Medicamentos.cantAmoxicilina-cantidad
        elif eleccion == tipoMedicamento.IBUPROFENO.getTipoMedicamento():
            Medicamentos.cantIbuprofeno=Medicamentos.cantIbuprofeno-cantidad
        else:
            Medicamentos.cantLagrimasArtificiales=Medicamentos.cantLagrimasArtificiales-cantidad
    
    

